update.go

```diff
@@ -70,39 +73,45 @@ func (uc *UpdateConfig) LoadFilesToDir(ver Version, dirPath string) error {
 		}
 		dirPath = filepath.Dir(exePath)
 	}
-	err := ver.getAssetsFilesContent(uc.ApplicationConfig, func(reader io.Reader, filename string) (err error) {
-		file, err := os.Create(filepath.Join(dirPath, filename))
-		if err != nil {
-			return err
-		}
-		defer func() {
-			tempErr := file.Close()
-			if err == nil {
-				err = tempErr
-			}
-		}()
-		_, err = io.Copy(file, reader)
-		return err
-	})
+	vfl := versionFilesLoader{
+		version:      ver,
+		updateConfig: uc,
+		getReplacementFileInfo: func(loadedFilename string) (ReplacementFile, error) {
+			return ReplacementFile{
+				FileName:           loadedFilename,
+				Mode:               ReplacementFileDefaultMode,
+				PreventFileLoading: false,
+			}, nil
+		},
+		destDir:             dirPath,
+		createOrdinaryFiles: true,
+	}
+	assetsFilenames := vfl.version.getAssetsFilenames()
+	_, err := vfl.loadUpdateFilesFromSource(assetsFilenames)
 	return err
 }
 
 const ReplacementFileInfoUseDefaultOrExistedFilePerm = 9999
 const ReplacementFileDefaultMode = fs.FileMode(0644)
 
 type ReplacementFile struct {
-	FileName string
-	Mode     fs.FileMode // use ReplacementFileInfoUseDefaultOrExistedFilePerm to set default ReplacementFileDefaultMode or if file already exist, existed file permission
+	FileName           string
+	fileDir            string      // rel path to dest dir
```

Comment:
It took me a while to find it from code
```suggestion
	fileDir            string      // rel path to dest dir (inside the archive)
```

Reply:
Also, maybe rename to `relDir`? or something with `rel`